src/test/java/com/coherent/zipcode/zipCodeController/ZipCodePOST.java

```diff
@@ -0,0 +1,132 @@
+    private URIManager zipPostUri = new URIManager();
+    private ZipCodeController zipCodeController = new ZipCodeController();
+    private static final Logger logger = LoggerFactory.getLogger(ZipCodePOST.class);
+
+//    @ParameterizedTest
+//    @CsvSource({
+//            "TEST1,TEST2,TEST3",
+//            "TEST1,TEST4,TEST5",
+//    })
+
+    @DisplayName("Scenario 2 - check zip codes have been created")
+    @Test
+    public void shouldCreateZipCodeWithPOST() {
+        String body = "[\"TEST1\",\"TEST2\",\"TEST3\"]";
+        CloseableHttpResponse postZipCodesResponse = zipCodeController.postZipCodes(body);
+        final int statusCode = postZipCodesResponse.getStatusLine().getStatusCode();
+
+        CloseableHttpResponse getZipCodes = zipCodeController.getZipCodes();
+
+        String responseBody = null;
+        try {
+            responseBody = EntityUtils.toString(getZipCodes.getEntity());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        logger.info("Get zip codes response body: {}", responseBody);
+        logger.info("Status code: {}", postZipCodesResponse.getStatusLine());
+
+        String finalResponseBody = responseBody;
+        assertAll("Zip Code POST Response",
+                () -> assertEquals(HttpStatus.SC_CREATED, statusCode),
+                () -> assertTrue(finalResponseBody.contains("TEST1"), "Expected value 'TEST1' not found on the server"),
+                () -> assertTrue(finalResponseBody.contains("TEST2"), "Expected value 'TEST2' not found on the server"),
+                () -> assertTrue(finalResponseBody.contains("TEST3"), "Expected value 'TEST3' not found on the server")
+
+        );
+        //BUGS:
+        //Scenario 2: No bugs
+    }
+
+    @DisplayName("Scenario 3 - check there are no duplicates for available zipcodes")
+    @Test
+    public void shouldCreateNoDuplicatesForAvailableZipCodes() {
+        String body = "[\"ABCDE\",\"TEST4\"]";
+        CloseableHttpResponse postZipCodesResponse = zipCodeController.postZipCodes(body);
+        final int statusCode = postZipCodesResponse.getStatusLine().getStatusCode();
+
+        CloseableHttpResponse getZipCodes = zipCodeController.getZipCodes();
+
+        String responseBody = null;
+        try {
+            responseBody = EntityUtils.toString(getZipCodes.getEntity());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        logger.info("Get zip codes response body: {}", responseBody);
+        logger.info("Status code: {}", postZipCodesResponse.getStatusLine());
+        String finalResponseBody = responseBody;
+        int occurences = StringUtils.countMatches(responseBody, "ABCDE");
+
+        assertAll("Zip Code POST Response",
+                () -> assertEquals(HttpStatus.SC_CREATED, statusCode),
+                () -> assertEquals(1, occurences, "Found duplicates for available zip codes: 'ABCDE'")
```

Comment:
As another option  you can save result from backend to the Set and than compare size

Reply:
It would be great to discuss this approach